watson-developer-cloud/discovery-components
# feat: identify Top Entities facet values by Type

#### What do these changes do/fix?
These changes add in a call with a nested aggregation query to get back the type for entities facets so we can use it to identify/group these facets' facet values by type. 

Any entities facet will be displayed with its facet values organized by type (I'm using `category` in the code to distinguish from types), so types like `Company`, `Person`, `Organization`, etc. On initial load, only the types will be shown, collapsed, beneath the entities facet's label. By clicking on them, they will expand to show their facet values. If there are more than 5 facet values, a `Show more` button will appear that will reveal the rest of the facet values if clicked. The facet values can be selected and cleared like any other facet. 

See screenshot here:
<img width="383" alt="Screen Shot 2020-04-03 at 5 18 22 PM" src="https://user-images.githubusercontent.com/6287012/78703917-0841fc80-78d9-11ea-81dc-1ba4f1756fd5.png">

#### How do you test/verify these changes?
You can link to Tooling and test in a workspace there. Try out some searches on a workspace with an entities facet and see that the types show up as expected. Play around with expanding/collapsing categories and facet values within categories. You can also test out in the example app with some searches there on a project that has an entities facet (Sample Project has a `Top Entities` by default).

You can also check it out in Storybook.

Also check out the tests that I added, and confirm they're testing everything we'd want to.

#### Have you documented your changes (if necessary)?
An example is displayed in Storybook for `SearchFacets` component.

#### Are there any breaking changes included in this pull request?
No
